### DISCUSSION TITLE
How to use 'Fake Ticker' with Caffeine?

### DISCUSSION THREAD

**Discussion by brent-holland** [ben-manes/caffeine#810] [Q&A] [answered]
In your discussion of testing [here](https://github.com/ben-manes/caffeine/wiki/Testing) you suggest that we use guava-testing's 'FakeTicker' class.  Unfortunately, when I try to use this I get an error of incompatible types.  Your Caffeine.ticker method is looking for a caffeine.cache.Ticker and FakeTicker is a google.common.base.Ticker, can you please tell me how I can use a fake ticker with caffeine?  I didn't see any sort of constructor / builder / converter method that took a google ticker and transformed to a caffeine ticker but perhaps I've missed something.  Really need this as having to use a Thread.sleep in a unit test is a no no :)

Error: 
incompatible types: com.google.common.base.Ticker cannot be converted to com.github.benmanes.caffeine.cache.Ticker

**Comment by ben-manes** [comment, marked as answer]
You missed the method reference to adapt. =)

```
FakeTicker ticker = new FakeTicker(); // Guava's testlib
Cache<Key, Graph> cache = Caffeine.newBuilder()
    .expireAfterWrite(10, TimeUnit.MINUTES)
    .executor(Runnable::run)
    .ticker(ticker::read)
    .maximumSize(10)
    .build();
```

**Reply by brent-holland** [reply]
You're absolutely right I missed the ref to read.  Works great now and my UT doesn't take minutes to run.  Thanks so much!
